lib/modal/edit_profile_modal.dart

```diff
@@ -0,0 +1,160 @@
+                hintText: 'Password',
+                labelText: 'Password',
+              ),
+              controller: passwordController,
+              obscureText: true,
+            ),
+            DropdownSearch(
+              asyncItems: (String filter) async {
+                final locations = await api.fetchLocations();
+                return locations;
+              },
+              onChanged: (value) {
+                locationController.text = value.id.toString();
+              },
+              dropdownDecoratorProps: const DropDownDecoratorProps(
+                dropdownSearchDecoration: InputDecoration(
+                  labelText: "Location",
+                ),
+              ),
+              popupProps: const PopupProps.menu(
+                fit: FlexFit.loose,
+              ),
+            ),
+          ],
+        ),
+        actions: [
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+            children: [
+              ElevatedButton(
+                onPressed: () {
+                  setUserData(user);
+                },
+                style: ElevatedButton.styleFrom(
+                  backgroundColor: ThemeClass.darkTheme.secondaryHeaderColor,
+                  foregroundColor: ThemeClass.darkTheme.primaryColor,
+                ),
+                child: const Text('Reset'),
+              ),
+              ElevatedButton(
+                onPressed: () {
+                  final password = passwordController.text.isEmpty
+                      ? null
+                      : passwordController.text;
+                  final success = api.updateUser(
+                    userId: user['id'],
+                    firstName: firstNameController.text,
+                    lastName: lastNameController.text,
+                    email: emailController.text,
+                    phoneNumber: phoneController.text,
+                    locatioId: int.parse(locationController.text),
+                    password: password,
+                  );
+
+                  success.then(
+                    (value) {
+                      if (value) {
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          const SnackBar(
+                            content: Text('User updated successfully'),
+                          ),
+                        );
+                        if (callback != null) {
+                          callback();
+                        }
+                      } else {
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          const SnackBar(
+                            content: Text('User update failed'),
+                            backgroundColor: Colors.red,
+                          ),
+                        );
+                      }
+                    },
+                  );
+                },
+                child: const Text('Update'),
```

Comment:
Close modal on update and remove focus as to close keyboard

Reply:
Okay, will do